DEVELOPING.md

```diff
@@ -17,7 +32,131 @@ This project:
+
+There **is** a host application, so `MapViewIntegrationTestCase` fetches the existing window and view controller before adding the MapView to it.
+
+- These tests can be run on devices.
+- Tests will be skipped on simulators < iOS 13 because there's no valid Metal device. (iOS 13+ has a simulated Metal device.)
+
+### What happens if I run from `MapboxMaps`
+
+There is **no host application**, so `MapViewIntegrationTestCase` creates its own `UIWindow` and root view controller, before adding the MapView to it.
+
+- These tests cannot be run on devices.
+- Tests will be skipped on simulators < iOS 13 because there's no valid Metal device.
+- Tests will be skipped entirely on CircleCI because of the VM (no Metal). **So for CI purposes, integration tests should be run on AWS Device Farm via `MapboxTestHost`**
+
+As you can see the most useful case is to run within from the host application. Be aware that this has the potential for side-effects, since the application is not restarted for each test.
+
+### Base classes
+
+- `IntegrationTestCase` is a base class that fetches the access token either from the Info.plist or from `UserDefaults` (which will override). The Info.plist in `MapboxMaps` has been set up to use the variable `MAPBOX_ACCESS_TOKEN`. **Please add a `developer.xcconfig` in the root of the repo, that looks like:**
+
+    ```
+    MAPBOX_ACCESS_TOKEN = pk.myaccesstoken
+    ```
+
+    The CircleCI config file has been set up to create this.
+
+- `MapViewIntegrationTestCase` subclasses the above, and sets the `MapView`, and is the `MBXMapViewDelegate`. Closures are used to expose these callbacks to subclasses. Please don't add tests to this class.
+
+- `ExampleIntegrationTest` is an example of using the above, that sets a style then waits for the load to finish.
+
+## Application Testing
+
+### Making an Example
+
+* Check out this [doc](https://github.com/mapbox/mapbox-maps-ios/blob/main/Apps/Examples/README.md) to get more information about adding examples to our project.### Setting up a development environment
+
+This project runs on Apple's Metal rendering framework. Prerequisites for running the test app in simulators:
+
+- macOS 10.15 or later
+- Xcode 12.2+
+
+This project:
+
+1. requires [a valid ~/.netrc file](https://docs.mapbox.com/ios/maps/guides/install/#configure-credentials) to fetch dependencies.
+
+2. uses [SwiftLint](https://github.com/realm/SwiftLint) to enforce a style guide. Install it with `brew install swiftlint` and check for lint by running `swiftlint` in the root of the repository.
+
+3. uses [Swift Package Manager](https://github.com/apple/swift-package-manager) to manage dependencies for development. Cocapods is supported for *consuming* the SDK.
+
+4. uses [secret-shield](https://github.com/mapbox/secret-shield) to help block secrets such as access tokens from being exposed. Setup `secret-shield` by running:
+```
+npm install -g @mapbox/secret-shield
+scripts/install-pre-commit/install-pre-commit.sh
+```
```

Comment:
This feels redundant with a very similar section above. Is there a reason to duplicate it?